app/controllers/api/members_controller.rb

```diff
@@ -16,4 +22,13 @@ def create
   def member_params
     params.permit(:name, :email, :country, :postal)
   end
+
+  def member_validation
+    [
+      { name: 'email', data_type: 'email', required: true },
+      { name: 'country', data_type: 'country', required: false },
+      { name: 'postal', data_type: 'postal', required: false },
+      { name: 'name', data_type: 'text', required: true }
+    ]
+  end
```

Comment:
From a purist perspective, a controller should delegate any job that needs to be done to another class, and only care about what response should give according to the result of the operation that was done. It shouldn't be doing any work by itself. In the real world tho, we bend this rule as a compromise, because it doesn't make sense to create a service that has only a couple lines of logic. But I believe this controller has gone way beyond that, there are 12 LOC that are strictly related to the member creation process. Extracting it to a service will make the code clearer, separating responsibilities and make it easier to test.